cmd/telegraf/telegraf.go

```diff
@@ -271,7 +271,7 @@ func reloadLoop(stop chan struct{}, s service.Service) {
 		log.Printf("I! Tags enabled: %s", c.ListTags())
 
 		if *fPidfile != "" {
-			f, err := os.Create(*fPidfile)
+			f, err := os.OpenFile(*fPidfile, os.O_EXCL|os.O_CREATE|os.O_WRONLY, 0666)
```

Comment:
I don't think that the pidfile should be used as a process lock. There are quite a few references on how to handle pidfiles, none seem very canonical, this might be the best one to go with (uses `O_RDWR | O_CREAT`): http://www.man7.org/tlpi/code/online/dist/filelock/create_pid_file.c.html